File: client/src/app/data/features/tournaments/dtos/competition/tournament.dto.ts
import { paginationFilter } from "../common/pagination-filter.dto";

export interface CreateTournament {
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
}
export interface UpdateTournamentDto extends CreateTournament {
  id: number;
  tournamentStateId: number;
  stateId: number;
  isPublic?: boolean;
}

export interface TournamentQuery extends paginationFilter {
  name?: string;
  description?: string;
  startDate?: string;
  endDate?: string;
  tournamentStateId?: number;
}

export interface TournamentResponseDto {
  id: number;
  name: string;
  description: string;
  startDate: string;
  endDate: string;
  tournamentStateId: number;
  tournamentStateName: string;
  stateId: number;
  stateName: string;
  isPublic?: boolean;
}

export interface TournamentBracketDto {
    tournamentId: number;
    tournamentName: string;
    phases: PhaseBracketDto[];
}

export interface PhaseBracketDto {
    phaseId: number;
    phaseName: string;
    phaseOrder: number;
    matches: MatchBracketDto[];
}

export interface MatchBracketDto {
    matchId: number;
    player1Name: string;
    player2Name: string;
    scorePlayer1: number | null;
    scorePlayer2: number | null;
    isFinal: boolean;
}

File: api/PatoCup.Application/DTOs/Competition/TournamentDto.cs
using PatoCup.Domain.Entities.Competition;

namespace PatoCup.Application.DTOs.Competition
{
    public class CreateTournamentDto
    {
        public string Name { get; set; } = null!;
        public string Description { get; set; } = null!;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public List<Phase> Phases { get; set; } = new();
    }

    public class UpdateTournamentDto : CreateTournamentDto
    {
        public int Id { get; set; }
        public int TournamentStateId { get; set; }
        public int StateId { get; set; }
    }

    public class TournamentResponseDto
    {
        public int Id { get; set; }
        public string? Name { get; set; } = null!;
        public string? Description { get; set; } = null!;
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public int? TournamentStateId { get; set; }
        public string TournamentStateName { get; set; } = null!;
        public int StateId { get; set; }
        public string StateName { get; set; } = null!;
        public bool IsPublic { get; set; }
    }

    public class TournamentBracketDto
    {
        public int TournamentId { get; set; }
        public string TournamentName { get; set; } = null!;
        public List<PhaseBracketDto> Phases { get; set; } = new();
    }

    public class PhaseBracketDto
    {
        public int PhaseId { get; set; }
        public string PhaseName { get; set; } = null!;
        public int PhaseOrder { get; set; }
        public bool IsFinal { get; set; }
        public List<MatchBracketDto> Matches { get; set; } = new();
    }

    public class MatchBracketDto
    {
        public int MatchId { get; set; }
        public string Player1Name { get; set; }
        public string Player2Name { get; set; } 
        public int? ScorePlayer1 { get; set; }
        public int? ScorePlayer2 { get; set; }
    }

}